inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/mq/MessageQueueZoneSinkContext.java

```diff
@@ -268,4 +308,40 @@ public void processSendFail(BatchPackProfile currentRecord, String topic, long s
             currentRecord.fail();
         }
     }
+
+    private void appendIndex(Event event, boolean isSuccess) {
+        if (event == null) {
+            return;
+        }
+        if (mqConfig.getStatIntervalSec() <= 0) {
+            return;
+        }
+        // add monitor items base file storage
+        String topic = event.getHeaders().get(ConfigConstants.TOPIC_KEY);
```

Comment:
These logic can implement in the user-defined MQ Handler.

Reply:
This is not MQ specific but message protocol specific, which is handled by common context logic.